hnn_core/tests/test_io.py

```diff
@@ -0,0 +1,354 @@
+    (net1_hard_change_drive.external_drives['evdist1']['weights_ampa']
+     ['L2_basket']) = 0
+    assert net1_hard_change_drive != net1
+
+
+def test_eq_conn(jones_2009_network):
+    net1 = jones_2009_network
+
+    # Check a change in connectivity
+    net1_clear_conn = net1.copy()
+    net1_clear_conn.clear_connectivity()
+    assert net1_clear_conn != net1
+
+    # Hardwired change in connectivity attribute
+    net1_hard_change_conn = net1.copy()
+    net1_hard_change_conn.connectivity[0]['gid_pairs'] = {}
+    assert net1_hard_change_conn != net1
+
+    # Hardwired change in connectivity nc_dict
+    net1_hard_change_conn = net1.copy()
+    net1_hard_change_conn.connectivity[0]['nc_dict']['A_weight'] = 0
+    assert net1_hard_change_conn != net1
+
+    # Check edge case, same number of connections, different replicate in conn
+    net1_alt_conn1 = net1.copy()
+    net1_alt_conn2 = net1.copy()
+    l_conn = net1_alt_conn1.connectivity
+    l_conn_rep_start = [l_conn[0]] + l_conn
+    l_conn_rep_end = l_conn + [l_conn[-1]]
+    net1_alt_conn1.connectivity = l_conn_rep_start
+    net1_alt_conn2.connectivity = l_conn_rep_end
+    assert net1 != net1_alt_conn1
+    assert net1_alt_conn1 == net1_alt_conn1
+    assert net1_alt_conn1 != net1_alt_conn2
+
+
+def test_write_network(tmp_path, jones_2009_network):
+    """ Tests that a hdf5 file is written """
+
+    # Check no file is already written
+    path_out = tmp_path / 'net.hdf5'
+    assert not path_out.is_file()
+
+    # Write network check
+    jones_2009_network.write(path_out)
+    assert path_out.is_file()
+
+    # Overwrite network check
+    last_mod_time1 = path_out.stat().st_mtime
+    jones_2009_network.write(path_out)
+    last_mod_time2 = path_out.stat().st_mtime
+    assert last_mod_time1 < last_mod_time2
+
+    # No overwrite check
+    with pytest.raises(FileExistsError,
+                       match="File already exists at path "):
+        jones_2009_network.write(path_out, overwrite=False)
+
+
+def test_write_network_no_output(tmp_path, jones_2009_network):
+    net = jones_2009_network.copy()
+    path_out = tmp_path / 'net.hdf5'
+
+    # Simulate and check for output
+    simulate_dipole(net, tstop=2, n_trials=1, dt=0.5)
+    assert np.array_equal(net.rec_arrays['el1'].times,
+                          [0.0, 0.5, 1.0, 1.5, 2.0]
+                          )
+
+    # Write to file
+    net.write(path_out, write_output=False)
+
+    # Read in file and check no outputs were written
+    hdf5_read = read_hdf5(path_out)
+    assert not any([bool(val['times'])
+                    for val in hdf5_read['rec_arrays'].values()]
+                   )
+    assert not any([bool(val['voltages'])
+                    for val in hdf5_read['rec_arrays'].values()]
+                   )
+    assert not any([bool(val['events'])
+                    for val in hdf5_read['external_drives'].values()]
+                   )
+
+
+def test_cell_response_to_dict(jones_2009_network):
```

Comment:
docstrings are missing in the test functions ... they are used by `pytest` when used in verbose mode. @gtdang can we add one-line docstrings to the test functions?